src/Microsoft.AspNet.Authentication/Events/SigningInContext.cs

```diff
@@ -25,6 +28,13 @@ public abstract class ReturnEndpointContext : EndpointContext
         public ClaimsPrincipal Principal { get; set; }
         public AuthenticationProperties Properties { get; set; }
 
+        public bool IsRequestCompleted { get; private set; }
+
+        public void RequestCompleted()
+        {
+            IsRequestCompleted = true;
```

Comment:
Yeah those are all kind of gross :smile: I'm ok keeping them as-is - I get why we're using this pattern. It's more of a "command" as opposed to a traditional method.